src/DateUtils.ts

```diff
@@ -175,7 +175,7 @@ function withinCurrentYear(prevDate: Date, nextDate: Date): boolean {
     return prevDate.getFullYear() === nextDate.getFullYear();
 }
 
-export function wantsDateSeparator(prevEventDate: Date, nextEventDate: Date): boolean {
+export function wantsDateSeparator(prevEventDate: Date | undefined, nextEventDate: Date | undefined): boolean {
```

Comment:
This is a bit weird, tbh but since the function already checks fall them being falsy, let's do it this way

Reply:
Definitely weird, but I figured updating the type annotations was less disruptive than changing how it works.